Framework/CurveFitting/src/HistogramDomainCreator.cpp

```diff
@@ -63,8 +63,8 @@ void HistogramDomainCreator::createDomain(
   }
 
   // set the data to fit to
-  const Mantid::MantidVec &Y = m_matrixWorkspace->readY(m_workspaceIndex);
-  const Mantid::MantidVec &E = m_matrixWorkspace->readE(m_workspaceIndex);
+  const auto &Y = m_matrixWorkspace->y(m_workspaceIndex);
+  const auto &E = m_matrixWorkspace->e(m_workspaceIndex);
```

Comment:
Below there is a code block

```cpp
     if (isDistribution) {
       // If workspace is a distribution, convert data to histogram
       auto dx = X[i + 1] - X[i];
       y *= dx;
       error *= dx;
}
```

`Histogram` can do that automatically: Instead of using `y` and `e`, use `counts` and `countStandardDeviations` here.